Apache-Beam-Jobs.md

```diff
@@ -227,6 +229,257 @@ error_pcoll = (
 
 `Runner`s provide the `run()` method used to visit every node (`PValue`) in the pipeline's DAG by executing the edges (`PTransform`s) to compute their values.  At Oppia, we use `DataflowRunner` to have our `Pipeline`s run on the [Google Cloud Dataflow service](https://cloud.google.com/dataflow).
 
+## Writing Apache Beam Jobs
+
+For this section, we'll walk through the steps of implementing a job by writing one: `CountExplorationStatesJob`.
+
+Here's the DAG we'll try to implement:
+
+    .--------------. Count states .--------. Sum .-------.
+    | Explorations | -----------> | Counts | --> | Total |
+    '--------------'              '--------'     '-------'
+
+### 1. Subclass the `base_jobs.JobBase` class and override the `run()` method
+
+Make sure your job class name is clear and concise, because the name is presented to release coordinators:
+
+<!-- TODO(#11475): Replace this screenshot with one that has job names which follow our naming convention. -->
+![Screenshot of the Release Coordinator Page with the list of jobs visible](https://user-images.githubusercontent.com/5094060/135092574-0bff536e-1b58-4b38-9358-c26f9096c8a3.png)
+
+Job names should follow the convention: `<Verb><Noun>Job`.
+
+-   For example:
+
+    ```python
+    class WeeklyDashboardStatsComputationJob(base_jobs.JobBase):
+        """BAD: Name does not begin with a verb."""
+
+        def run(self):
+            ...
+
+
+    class ComputeStatsJob(base_jobs.JobBase):
+        """BAD: Unclear what kind of stats are being computed."""
+
+        def run(self):
+            ...
+
+
+    class CountExplorationStatesJob(base_jobs.JobBase):
+        """GOOD: Name starts with a verb and "exploration states" is unambiguous."""
+
+        def run(self):
+            ...
+    ```
+
+Module names should follow the convention: `<noun>_<operation>_jobs.py`.
+
+-   For example:
+    * `blog_validation_jobs.py`
+    * `dashboard_stats_computation_jobs.py`
+    * `exploration_indexing_jobs.py`
+    * `exploration_stats_regeneration_jobs.py`
+    * `model_validation_jobs.py`
+
+    However, you should always prefer placing jobs in pre-existing modules if an appropriate one already exists.
+
+For this example, we will write our job in the module: `core/jobs/batch_jobs/exploration_inspection_jobs.py`.
+
+### 2. Override the `run()` method to operate on `self.pipeline`
+
+As illustrated in the Architecture section, jobs are organized by `Pipeline`s, `PTransform`s, and `PCollection`s. Thus, when writing your job, **you must build _off_ of your pipeline** in order to construct the DAG. We can represent this by adding a special `Pipeline` node signifying the _true_ beginning of the DAG:
```

Comment:
I don't really know what "my pipeline" is. Can you explain conceptually what's happening here and why the pipeline node is the true beginning of the DAG? Does my pipeline already exist or is it a new thing that I am constructing as part of implementing the job? (The current doc suggests the former.)

Reply:
Done.